arches/app/models/resource.py

```diff
@@ -146,36 +148,36 @@ def get_descriptor(self, descriptor, context):
             except KeyError:
                 pass
 
-        self.calculate_descriptors(context=context)
-        return self.descriptors[language][descriptor]
-
-    def calculate_descriptors(self, descriptors=("name", "description", "map_popup"), context=None):
+    def save_descriptors(self, descriptors=("name", "description", "map_popup"), context=None):
         """
         descriptors -- iterator with descriptors to be calculated
-        context -- Dictionary which may have:
-            language -- Language code in which the descriptor should be returned (e.g. 'en').
-                This occurs when handling concept values.
-            any key:value pairs needed to control the behavior of a custom descriptor function
+        context -- Dictionary with any key:value pairs needed to control the behavior of a custom descriptor function
 
         """
 
-        language = self.get_descriptor_language(context)
-
-        if not self.descriptor_function:
-            self.descriptor_function = models.FunctionXGraph.objects.filter(
-                graph_id=self.graph_id, function__functiontype="primarydescriptors"
-            ).select_related("function")
-
-        for descriptor in descriptors:
-            if len(self.descriptor_function) == 1:
-                module = self.descriptor_function[0].function.get_class_module()()
-                self.descriptors[language][descriptor] = module.get_primary_descriptor_from_nodes(
-                    self, self.descriptor_function[0].config["descriptor_types"][descriptor], context
-                )
-                if descriptor == "name" and self.descriptors[language][descriptor] is not None:
-                    self.name[language] = self.descriptors[language][descriptor]
+        for lang in settings.LANGUAGES:
+            language = self.get_descriptor_language({"language":lang[0]})
+            if context:
+                context["language"] = language
             else:
-                self.descriptors[language][descriptor] = None
+                context = {"language":language}
+            if not self.descriptor_function:
+                self.descriptor_function = models.FunctionXGraph.objects.filter(
+                    graph_id=self.graph_id, function__functiontype="primarydescriptors"
+                ).select_related("function")
```

Comment:
This should be deindented so we don't multiply the number of queries by the number of languages.

Spotted when applying the test case from #10316 to this branch.

Reply:
ah -- this isn't terrible because it's only making that query if there's not already a descriptor function, and the logic sets the descriptor function.

But it is an improvement to readability to happy to move it out of the block

Reply:
ah -- this isn't terrible because it's only making that query if there's not already a descriptor function, and the logic sets the descriptor function.

But it is an improvement to readability to happy to move it out of the block

Reply:
Interesting, I was definitely getting 10 queries instead of 2:
```py
(venv311) jwalls@Jacobs-MacBook-Pro arches % python3 manage.py test tests.models.resource_test.ResourceTests.test_recalculate_descriptors_one_query_for_descriptor_function --settings="tests.test_settings"
Found 1 test(s).
F
======================================================================
FAIL: test_recalculate_descriptors_one_query_for_descriptor_function (tests.models.resource_test.ResourceTests.test_recalculate_descriptors_one_query_for_descriptor_function)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jwalls/prj/arches/tests/models/resource_test.py", line 254, in test_recalculate_descriptors_one_query_for_descriptor_function
    self.assertEqual(len(function_x_graph_selects), 1)
AssertionError: 10 != 1

----------------------------------------------------------------------
Ran 1 test in 2.091s

FAILED (failures=1)
```

Reply:
I can take a look to see if the test setup is not realistic 👍

Reply:
Ah, what I think is is this: the check against `self.descriptor_function` was testing truth value, not is None, so even though the logic "sets" the descriptor_function, it didn't help in the case of a resource without a descriptor function. But we'll be fixing this soon anyway in #10316 👍